rest-dashboard.go

```diff
@@ -271,31 +312,20 @@ func (r *Client) SetDashboard(ctx context.Context, board Board, params SetDashbo
 	return resp, nil
 }
 
-// SetRawDashboard updates existing dashboard or creates a new one.
-// Contrary to SetDashboard() it accepts raw JSON instead of Board structure.
-// Grafana only can create or update a dashboard in a database. File dashboards
-// may be only loaded with HTTP API but not created or updated.
-//
-// Reflects POST /api/dashboards/db API call.
-func (r *Client) SetRawDashboard(ctx context.Context, raw []byte) (StatusMessage, error) {
+//SetRawDashboardWithParam sends the serialized along with request parameters
+func (r *Client) SetRawDashboardWithParam(ctx context.Context, request RawBoardRequest) (StatusMessage, error) {
 	var (
 		rawResp []byte
 		resp    StatusMessage
 		code    int
 		err     error
-		buf     bytes.Buffer
-		plain   = make(map[string]interface{})
 	)
-	if err = json.Unmarshal(raw, &plain); err != nil {
-		return StatusMessage{}, err
+	raw, err := json.Marshal(request)
+
+	if err != nil {
+		panic(err)
```

Comment:
Don't ever, ever `panic()` in a library. We don't want to crash our users' programs. Please return `StatusMessage{}, err`

Reply:
Good point, Done